ui_framework/src/components/button/button.js

```diff
@@ -71,6 +71,9 @@ const KuiButton = ({
   children,
   ...rest
 }) => {
+  if (!children && !rest['aria-label'] && !rest['aria-labelledby']) {
+    console.error('Specify aria-label or aria-labelledby on <KuiButton/> if it does not have children.');
+  }
```

Comment:
What would you say to replacing this with a custom prop-type checker?

```jsx
const accessibleIconButton = (props, propName, componentName) => {
  if (props.children) {
    return;
  }

  if (props['aria-label']) {
    return;
  }

  if (props['aria-labelledby']) {
    return;
  }

  throw new Error(
    `${componentName} requires aria-label or aria-labelledby to be specified if it does not have children. ` +
    `This is because we're assuming you're creating an icon-only button, which is screen-reader-inaccessible.`
  );
};

const KuiButton = ({
  isLoading,
  iconPosition = DEFAULT_ICON_POSITION,
  className,
  buttonType,
  icon,
  children,
  ...rest
}) => {
  return (
    <button
      className={getClassName({
        className,
        buttonType,
        hasIcon: icon || isLoading,
      })}
      {...rest}
    >
      <ContentWithIcon
        icon={icon}
        iconPosition={iconPosition}
        isLoading={isLoading}
      >
        {children}
      </ContentWithIcon>
    </button>
  );
};

KuiButton.propTypes = {
  icon: PropTypes.node,
  iconPosition: PropTypes.oneOf(ICON_POSITIONS),
  children: PropTypes.node,
  isLoading: PropTypes.bool,
  buttonType: PropTypes.oneOf(BUTTON_TYPES),
  className: PropTypes.string,
  'aria-label': accessibleIconButton,
};
```

This will give us an error like this:

![image](https://user-images.githubusercontent.com/1238659/30676245-ed807678-9e39-11e7-8e6d-30b058689c8c.png)

Reply:
That is of course the way better option. Sorry I never thought about using `propTypes` for checking conditions between multiple properties.